controllers/internalrequest/adapter.go

```diff
@@ -136,11 +136,10 @@ func (a *Adapter) EnsureRequestIsAllowed() (controller.OperationResult, error) {
 		}
 	}
 
-	patch := client.MergeFrom(a.internalRequest.DeepCopy())
```

Comment:
FYI - From previous experience with this, not using the DeepCopy() might cause a reconciler error  (that one saying that the object was modified), so the copy + patch solution used here. Anyhow I see e2e running at the moment I am writing, if it passes I have no objection with the change and will approve.